src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,352 @@
+- `omega::Union{Float64, Nothing}`, The exponential distrbution parameter, if not given is determined for sampling methods.
+- `eta::Float64`, A parameter for adjusting the conservativeness of the distribution, higher value means a less conservative
+  estimate. By default, this is set to one.
+"""
+mutable struct DistInfo
+    sampler::Union{DataType, Nothing}
+    max_dimension::Int64
+    block_dimension::Int64
+    sigma2::Union{Float64, Nothing}
+    omega::Union{Float64, Nothing}
+    eta::Float64
+end
+
+"""
+    LSLogMA <: LinSysSolverLog
+
+A mutable structure that stores information about a randomized linear solver's behavior.
+    The log assumes that only a random block of full linear system is available for processing. 
+    The goal of this log is usually for all use cases as it acts as a cheap approximation of the 
+    full residual.
+
+# Fields
+- `collection_rate::Int64`, the frequency with which to record information to append to the
+    remaining fields, starting with the initialization (i.e., iteration 0).
+- `ma_info::MAInfo`, structure that holds information relevant only to moving average, 
+   like the two choices of moving average widths (lambda1 and lambda2), the current moving average
+   width (lambda), a flag to indicate which moving average regime, between lambda1 and lambda2, is 
+   being used, a vector containing the values to be averaged (res_window), and an indicator of where 
+   the next value should be placed (idx).
+- `resid_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `iota_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `width_hist::Vector{Int64}`, data structure that contains the widths used in the moving average 
+   calculation at each iteration.
+- `resid_norm::Function`, a function that accepts a single vector argument and returns a
+    scalar. Used to compute the residual size.
+- `iterations::Int64`, the number of iterations of the solver.
+- `converged::Bool`, a flag to indicate whether the system has converged by some measure.
+- `true_res`, a boolean indicating if we want the true residual computed instead of approximate.
+- `dist_info::DistInfo`, a structure that stores information about the sub-Exponential distribution.
+# Constructors
+- Calling `LSLogMA()` sets `collection_rate = 1`,  `lambda1 = 1`,
+    `lambda2 = 30`, `resid_hist = Float64[]`, `iota_hist = Float64[]`, `width_hist = Int64[]`, 
+    `resid_norm = norm` (Euclidean norm), `iterations = -1`, `converged = false`, 
+    `sampler = LinSysVecRowDetermCyclic`, `max_dimension = 0`, `sigma2 = nothing`, `omega = nothing`,
+    `eta = 1`, and `true_res = false`. The user can specify their own values of lambda1, lambda2, sigma2, omega, and eta using 
+    key word arguments as inputs into the constructor.
```

Comment:
A lot of these arguments are for setting the defaults of sub-`struct`s inside of `LSLogMA`. It might be better to create default constructors for these sub-`struct`s and specify that they are set to their default constructor values instead of doing it here. Otherwise, you need to update the docstring to indicate to which `struct`s some of these arguments belong for clarity. It is up to you.

I would also make a note for creating guidance on how to do this in v0.2. I think we will likely need to modify the blue style guide for our purposes.